inc/abstracttarget.class.php

```diff
@@ -1501,6 +1512,79 @@ protected function showLocationSettings($rand) {
       echo '</tr>';
    }
 
+   protected function showValidationSettings($rand) {
+      echo '<tr>';
+
+      // Setting label
+      echo '<td width="15%">' . __('Validation') . '</td>';
+
+      // Possible values
+      echo '<td width="45%">';
+      Dropdown::showFromArray('validation_rule', static::getEnumValidationRule(), [
+         'value'     => $this->fields['validation_rule'],
+         'on_change' => "plugin_formcreator_change_validation($rand)",
+         'rand'      => $rand
+      ]);
+      echo Html::scriptBlock("plugin_formcreator_change_validation($rand)");
+      echo '</td>';
+
+      // Hidden secondary labels, displayed according to the user main choice
+      echo '<td width="15%">';
+      $titles = [
+         'validation_specific_title'       => __('Approver'),                 // VALIDATION_RULE_SPECIFIC_USER_OR_GROUP
+         'validation_from_question_title'  => __('Question', 'formcreator'),  // VALIDATION_RULE_ANSWER_USER
+      ];
+      foreach ($titles as $id => $title) {
+         echo "<span id='$id' style='display: none'>$title</span>";
+      }
+      echo '</td>';
+
+      // Hidden secondary values, displayed according to the user main choice
+      echo '<td width="25%">';
+
+      // VALIDATION_RULE_SPECIFIC_USER_OR_GROUP
+      echo '<div id="validation_specific" style="display: none">';
+      $validation_dropdown_params = [
+         'name' => 'validation_specific'
+      ];
+      $validation_data = json_decode($this->fields['validation_question'], true);
+      if (!is_null($validation_data) && isset($validation_data['type'])) {
```

Comment:
when ```isset($validation_data['type'])``` is true, ```is_null($validation_data) ``` should be always false,  then this is redundant

```suggestion
      if (isset($validation_data['type'])) {
```